llvm/lib/Target/AMDGPU/AMDGPULowerModuleLDSPass.cpp

```diff
@@ -327,16 +327,18 @@ class AMDGPULowerModuleLDS {
     return convertUsersOfConstantsToInstructions(LDSGlobals);
   }
 
+  std::optional<bool> HasAbsoluteGVs;
```

Comment:
This is only used by one function, make it a local variable and reinstate the static annotations. Minimising scope of mutable state is important.